src/main/java/org/ebyhr/trino/storage/operator/ParquetPlugin.java

```diff
@@ -0,0 +1,289 @@
+    }
+
+    private ParquetReader getReader(File file)
+            throws IOException
+    {
+        ParquetReaderOptions options = new ParquetReaderOptions();
+        ParquetDataSource dataSource;
+        dataSource = new FileParquetDataSource(file, options);
+        ParquetMetadata parquetMetadata = MetadataReader.readFooter(dataSource, Optional.empty());
+        FileMetadata fileMetaData = parquetMetadata.getFileMetaData();
+        MessageColumnIO messageColumnIO = getColumnIO(fileMetaData.getSchema(), fileMetaData.getSchema());
+        ImmutableList.Builder<Column> columnFields = ImmutableList.builder();
+        for (org.apache.parquet.schema.Type type : fileMetaData.getSchema().getFields()) {
+            columnFields.add(new Column(
+                    messageColumnIO.getName(),
+                    constructField(
+                            fromParquetType(type),
+                            lookupColumnByName(messageColumnIO, type.getName()))
+                            .orElseThrow()));
+        }
+        Map<List<String>, ColumnDescriptor> descriptorsByPath = getDescriptors(fileMetaData.getSchema(), fileMetaData.getSchema());
+        long nextStart = 0;
+        ImmutableList.Builder<RowGroupInfo> rowGroupInfoBuilder = ImmutableList.builder();
+        for (BlockMetadata block : parquetMetadata.getBlocks()) {
+            rowGroupInfoBuilder.add(
+                    new RowGroupInfo(
+                            createPrunedColumnsMetadata(
+                                    block,
+                                    dataSource.getId(),
+                                    descriptorsByPath),
+                            nextStart,
+                            Optional.empty()));
+            nextStart += block.rowCount();
+        }
+        return new ParquetReader(
+                Optional.ofNullable(fileMetaData.getCreatedBy()),
+                columnFields.build(),
+                rowGroupInfoBuilder.build(),
+                dataSource,
+                UTC,
+                newSimpleAggregatedMemoryContext(),
+                new ParquetReaderOptions(),
+                exception -> {
+                    throwIfUnchecked(exception);
+                    return new RuntimeException(exception);
+                },
+                Optional.empty(),
+                Optional.empty());
+    }
+
+    private void handleException(File file, ParquetDataSource dataSource, Exception e)
+    {
+        try {
+            if (dataSource != null) {
+                dataSource.close();
+            }
+        }
+        catch (IOException ignored) {
+        }
+        if (e instanceof TrinoException) {
+            throw (TrinoException) e;
+        }
+        if (e instanceof ParquetCorruptionException) {
+            throw new RuntimeException(e);
+            // todo: when we integrate it into trino we can use trino's dependency...
+            // throw new TrinoException(HIVE_BAD_DATA, e);
```

Comment:
You can still throw `TrinoException` here, maybe with a more generic code.

Reply:
Is there a way we can import `io.trino.plugin.hive`? I tried importing 

```
        <dependency>
            <groupId>io.trino</groupId>
            <artifactId>trino-hive</artifactId>
            <version>${dep.trino.version}</version>
        </dependency>
````
but that doesn't seem like it

Reply:
We should not import the whole plugin here, so it's ok as is.